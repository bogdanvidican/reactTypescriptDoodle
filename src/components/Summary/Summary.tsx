import * as React from 'react';
import IPicture from '../Cards';
import { PicGrid } from '../PicGrid';
import Button from '@material-ui/core/Button';

export interface Props {
  pictures: IPicture[],
  restart: () => void,
}

const KEYS = {
  RATING: 'rating',
  SKIPPED: 'skipped',
}

export const Summary = ({ pictures, restart }: Props) => {
  const getPics = (key: string, value: boolean) => {
    return pictures.filter((pic: IPicture) => pic[key] === value)
  }

  const likedPics = getPics(KEYS.RATING, true)
  const disLikedPics = getPics(KEYS.RATING, false)
  const skippedPics = getPics(KEYS.SKIPPED, true)

  return (
    <div>
      <Button
        variant="outlined"
        color="secondary"
        onClick={restart}
      >
        Restart
      </Button>
      {!!(likedPics.length) && <PicGrid type="liked" pictures={likedPics} />}
      {!!(disLikedPics.length) && <PicGrid type="disliked" pictures={disLikedPics} />}
      {!!(skippedPics.length) && <PicGrid type="skipped" pictures={skippedPics} />}
    </div>
  )
}
