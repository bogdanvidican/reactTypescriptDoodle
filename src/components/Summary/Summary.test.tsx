import * as React from 'react';
import { shallow } from 'enzyme';
import { Summary, Props } from './Summary';
import { Button } from '@material-ui/core';

const props: Props = {
  pictures: [
    {
      url: 'https://some.test.url_1',
      thumb: 'https://some.test.thumb_1',
      title: 'some test title_1',
    },
    {
      url: 'https://some.test.url_1',
      thumb: 'https://some.test.thumb_1',
      title: 'some test title_1',
    },
    {
      url: 'https://some.test.url_2',
      thumb: 'https://some.test.thumb_2',
      title: 'some test title_2',
    },
    {
      url: 'https://some.test.url_3',
      thumb: 'https://some.test.thumb_3',
      title: 'some test title_3',
    },{
      url: 'https://some.test.url_4',
      thumb: 'https://some.test.thumb_4',
      title: 'some test title_4',
    }
  ],
  restart: jest.fn(),
};

let wrapper: any;
let button: any;

beforeAll(() => {
  wrapper = shallow(<Summary {...props} />);
  button = wrapper.find(Button);
})

describe('<Summary />', () => {
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })
  it('should render a restart button', () => {
    expect(button).toBeTruthy();
  })
  it('should trigger callback on button click', () => {
    button.simulate('click');
    expect(props.restart).toHaveBeenCalled();
  });
});
