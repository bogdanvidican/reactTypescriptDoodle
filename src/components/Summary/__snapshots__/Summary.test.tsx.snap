// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Summary /> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    pictures={
        Array [
            Object {
              "thumb": "https://some.test.thumb_1",
              "title": "some test title_1",
              "url": "https://some.test.url_1",
            },
            Object {
              "thumb": "https://some.test.thumb_1",
              "title": "some test title_1",
              "url": "https://some.test.url_1",
            },
            Object {
              "thumb": "https://some.test.thumb_2",
              "title": "some test title_2",
              "url": "https://some.test.url_2",
            },
            Object {
              "thumb": "https://some.test.thumb_3",
              "title": "some test title_3",
              "url": "https://some.test.url_3",
            },
            Object {
              "thumb": "https://some.test.thumb_4",
              "title": "some test title_4",
              "url": "https://some.test.url_4",
            },
          ]
    }
    restart={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WithStyles(Button)
          color="secondary"
          onClick={[Function]}
          variant="outlined"
>
          Restart
</WithStyles(Button)>,
        false,
        false,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Restart",
          "color": "secondary",
          "onClick": [Function],
          "variant": "outlined",
        },
        "ref": null,
        "rendered": "Restart",
        "type": [Function],
      },
      false,
      false,
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WithStyles(Button)
            color="secondary"
            onClick={[Function]}
            variant="outlined"
>
            Restart
</WithStyles(Button)>,
          false,
          false,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Restart",
            "color": "secondary",
            "onClick": [Function],
            "variant": "outlined",
          },
          "ref": null,
          "rendered": "Restart",
          "type": [Function],
        },
        false,
        false,
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
